h2/src/main/org/h2/expression/Function.java

```diff
@@ -1805,53 +1803,79 @@ private static int getDatePart(String part) {
         return p.intValue();
     }
 
-    private static Timestamp dateadd(String part, long count, Timestamp d) {
+    private static Value dateadd(String part, long count, Value v) {
         int field = getDatePart(part);
+        //v = v.convertTo(Value.TIMESTAMP);
+        if (field != MILLISECOND && count > Integer.MAX_VALUE || count < Integer.MIN_VALUE) {
```

Comment:
is this meant to be 
   (field != MILLISECOND && count > Integer.MAX_VALUE) || count < Integer.MIN_VALUE
or
  field != MILLISECOND && (count > Integer.MAX_VALUE || count < Integer.MIN_VALUE)
either way, some brackets would help